extensions/lifecycle/tasks/LifecycleUpdateTransitionTask.js

```diff
@@ -72,7 +70,8 @@ class LifecycleUpdateTransitionTask extends BackbeatTask {
         objMD.setLocation(location)
             .setDataStoreName(newLocationName)
             .setAmzStorageClass(newLocationName)
-            .setOriginOp('s3:LifecycleTransition');
+            .setOriginOp('s3:LifecycleTransition')
+            .setTransitionInProgress(false);
```

Comment:
This happens at the end of the transition, when the object has been moved to the new location.
It must be reset (similar to the end of the cold storage), to ensure the object becomes elligible to the "next" transition...

I think there should be a check already on the actual location of the object though (e.g. to avoid moving an object to the location where it is already), I will double check.

Reply:
Ok, found he protection, in [Arsenal's `getApplicableRules`](https://github.com/scality/arsenal/blob/acd13ff31b92b3d9469e8625c9a9f671d06a1ec2/lib/s3middleware/lifecycleHelpers/LifecycleUtils.ts):

```js
        // Do not transition to a location where the object is already stored.
        if (applicableRules.Transition
            && applicableRules.Transition.StorageClass === metadata.StorageClass) {
            applicableRules.Transition = undefined;
        }

        if (applicableRules.NoncurrentVersionTransition
            && applicableRules.NoncurrentVersionTransition.StorageClass === metadata.StorageClass) {
            applicableRules.NoncurrentVersionTransition = undefined;
        }
```

This however expects the `StorageClass` to match the name of the current location, which is not the case our case: the storage class will be STANDARD most of the time, i.e. except for cold objects (where it will indeed be name of the location)